src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -454,11 +454,11 @@ private function resolveServices($value)
             $value = array_map(array($this, 'resolveServices'), $value);
         } elseif (is_string($value) &&  0 === strpos($value, '@=')) {
             return new Expression(substr($value, 2));
-        } elseif (is_string($value) &&  0 === strpos($value, '@')) {
-            if (0 === strpos($value, '@@')) {
+        } elseif (is_string($value) && $value[0] && '@' === $value[0]) {
+            if ('@' === $value[1]) {
```

Comment:
before this check you should handle the case of ``$value`` being the string ``@`` only (which should create a Reference to the service using the empty string as id)